### PR TITLE
[BE] Enable flake8-comprehension rule C417

### PR DESCRIPTION
**Pull request by Skylion007**
Enables flake8-comprehension rule C417. Ruff autogenerated these fixes to the codebase.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Similar to the other PR. These need to be carefully reviewed but not much benefit?

**Comment by Skylion007** [comment]
> Similar to the other PR. These need to be carefully reviewed but not much benefit?

Much smaller than the other PR, it offers some minor code cleanliness + performance benefits and can be easily linted. This design pattern is pretty rare in the codebase, so this just nudges new to code using a slightly better style, more pythonic, more performant style.

**Comment by albanD** [comment]
Do you have some more details on what is the difference between these two forms? If one is more performant as you suggest then they are doing different things. In which cases exactly do they diverge and we could be observing different behavior?

Sorry for the paranoia but given all the critical issues stemming for what seems to be simple refactor like the six removal, I think we should be a lot more careful with this kind of changes.

**Comment by Skylion007** [comment]
Sure, so `map` can be more performant if the function being evaluated is a bound function (and especially if its a builtin conversion function link (int, str, etc...). However, this checks looks  for unbound lambda, where it is better and more pythonic to use a generator comprehension since you need to call the map function and pass in this lambda which needs to be constructed and then applied to each element in a generator. 

Converting this to a generator comprehension also allows other flake8 checks to convert a generator inside of a `list,tuple,dict` function call to be the corresponding type of collection comprehension. Functionally, they are the same, but map has more overhead due to the unbound lambda and unnecessary function call return. The python engine can generate more efficient bytecode and applies further specific optimizations to the simpler generator comprehension. It also is easier to read and takes fewer characters.

All the flake8-comprehension checks were designed to be used in django which has a ton of very performance sensitive comprehensions so these comprehensions are only added to the plugin if they are more efficient (which can be measured by either benchmarking or looking directly the Python generated bytecode to see if it is simpler).

**Comment by Skylion007** [comment]
> Do you have some more details on what is the difference between these two forms? If one is more performant as you suggest then they are doing different things. In which cases exactly do they diverge and we could be observing different behavior?
> 
> Sorry for the paranoia but given all the critical issues stemming for what seems to be simple refactor like the six removal, I think we should be a lot more careful with this kind of changes.

Oh yeah, totally agree about the six removal, but the main issue is a many of those were sensitive functional changes that should have been more carefully looked at. (Especially the bytes vs sequence gotchas).

File: benchmarks/sparse/dlmc/utils.py [hunk 1]
```diff
@@ -39,9 +39,9 @@ def csr_to_coo(indices, indptr, shape):
 
 def load_sparse_matrix(path, device):
     with open(path, 'r') as file:
-        nrows, ncols, nnz = map(lambda el: int(el), file.readline().split(', '))
```
Comment (ezyang) [inline benchmarks/sparse/dlmc/utils.py:42]: how come the original author didn't just write int 🤔

**Review by ezyang — APPROVED** [review]
Approving because the diff was autogenerated

File: benchmarks/sparse/dlmc/utils.py (hunk 1 above)
Comment (Skylion007) [reply benchmarks/sparse/dlmc/utils.py:42]: Exactly haha

**Review by kit1980 — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**